addons/stock_calendar/models/procurement.py

```diff
@@ -52,13 +48,14 @@ def _get_purchase_order_date(self, cr, uid, procurement, company, schedule_date,
         """
         if procurement.next_purchase_date:
             return datetime.strptime(procurement.next_purchase_date, DEFAULT_SERVER_DATETIME_FORMAT)
-        return super(procurement_order, self)._get_purchase_order_date(cr, uid, procurement, company, schedule_date, context=context)
+        return super(ProcurementOrder, self)._get_purchase_order_date(cr, uid, procurement, company, schedule_date, context=context)
```

Comment:
In case of @api.v7 & @api.v8 all logic will be in @api.v8 only @api.v7 use only for calling @api.v8 method :o: